src/player.py

```diff
@@ -12,4 +15,30 @@ def take_item(self, item_name):
         else:
             self.inventory.append(available[0])
             self.room.remove_item(item_name)
-            return f'{item_name} successfully taken!\n'
\ No newline at end of file
+            return f'{item_name} successfully taken!\n'
+            available[0].on_take()
+
+    def drop_item(self, item_name):
+        available = [x for x in self.inventory if x.name == item_name]
+        if not available:
+            return "You don't have one of those!\n"
+        else:
+            self.room.add_item(available[0])
+
+            available[0].on_drop()
+
+            for i in range(len(self.inventory)):
+                if self.inventory[i].name == item_name:
```

Comment:
same here, you can do it without looping. you saw an example of this in yesterday's code.